### PR TITLE
KAFKA-3479: consumer metrics doc

### PR DESCRIPTION
**Pull request by coughman**
added new consumer metrics section
refactored common metrics into new section
updated TOC

### CODE REVIEW

**Comment by hachikuji** [comment]
@coughman Thanks for picking this up. I think we have some topic-level metrics for the new consumer as well which can be found in `Fetcher.FetchManagerMetrics`. Maybe they should go here also?

**Comment by coughman** [comment]
sure I'll add those in

**Comment by coughman** [comment]
@hachikuji doesn't this commit already contain the fetch manager metrics? https://github.com/coughman/kafka/commit/8c6ef282dddd68315d735f34d87fd93e21b50309
I'll update the section header to be "fetch manager"

**Comment by coughman** [comment]
added subsections for topic-level, global request, and global connection metrics

File: docs/ops.html [hunk 1]
```diff
@@ -935,6 +974,289 @@
...
+    </tr>
+    <tr>
+      <td>outgoing-byte-rate</td>
+      <td>The average number of outgoing bytes sent per second to all servers.</td>
+      <td>kafka.consumer:type=consumer-metrics,client-id=([-.\w]+)</td>
+    </tr>
+  </tbody>
+</table>
+
+<h5><a id="consumer_global_connection_monitoring" href="#consumer_global_connection_monitoring">Global Connection Metrics</a></h5>
```
Comment (hachikuji) [inline docs/ops.html]: These metrics seem to already be covered in the "Common monitoring metrics" section above.

Comment (coughman) [inline docs/ops.html]: Thanks for pointing out.  Removed duplicate section.

File: docs/ops.html [hunk 2]
```diff
@@ -678,6 +678,120 @@
...
+      <td>The average number of network operations (reads or writes) on all connections per second.</td>
+      <td>kafka.[producer|consumer|connect]:type=[producer|consumer|connect]-metrics,client-id=([-.\w]+)</td>
+    </tr>
+    <tr>
+      <td>outgoing-byte-rate</td>
+      <td>The average number of outgoing bytes sent per second to all servers.</td>
+      <td>kafka.[producer|consumer|connect]:type=[producer|consumer|connect]-metrics,client-id=([-.\w]+)</td>
+    </tr>
+    <tr>
+      <td>request-rate</td>
```
Comment (hachikuji) [inline docs/ops.html]: I think you may have accidentally copied some of these metrics twice. This one and a few others are already listed in this table.

File: docs/ops.html [hunk 3]
```diff
@@ -935,6 +974,236 @@
...
+    </tr>
+    <tr>
+      <td>records-consumed-rate</td>
+      <td>The average number of records consumed per second for a specific topic.</td>
+      <td>kafka.consumer:type=consumer-fetch-manager-metrics,client-id=([-.\w]+),topic=([-.\w]+)</td>
+    </tr>
+  </tbody>
+</table>
+
+<h5><a id="consumer_global_request_monitoring" href="#consumer_global_request_monitoring">Global Request Metrics</a></h5>
```
Comment (hachikuji) [inline docs/ops.html]: I think all of these metrics come from `Selector`, so they should really be in the common section above. In fact, it looks like all of them except `request-latency-avg` and `request-latency-max` are there already. Maybe we can remove this section and add those two metrics to the common table?

**Comment by coughman** [comment]
removed "global request section".  refactored into common node metrics

**Comment by ewencp** [comment]
@coughman I think @hachikuji's comment about duplicates still holds -- I see request-rate, request-size-avg, request-size-max, etc multiple times.

**Comment by coughman** [comment]
removed dupes

**Comment by ewencp** [comment]
LGTM, thanks!

**Pull request closed without merging**